iothub/device/src/DeviceClient.cs

```diff
@@ -613,12 +617,45 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         /// <summary>
         /// Releases the unmanaged resources used by the DeviceClient and optionally disposes of the managed resources.
         /// </summary>
+        /// <remarks>
+        /// The method <see cref="CloseAsync()"/> should be called before disposing.
+        /// </remarks>
         public void Dispose()
         {
             Dispose(true);
             GC.SuppressFinalize(this);
         }
 
+        /// <summary>
+        /// Disposes the client in an async way. See <see cref="IAsyncDisposable"/> for more information.
+        /// </summary>
+        /// <remarks>
+        /// Includes a call to <see cref="CloseAsync()"/>.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// await using var client = DeviceClient.CreateFromConnectionString(...);
```

Comment:
Does this mean if you use the `await` here, when time comes to dispose, the `DisposeAsync` will be called automatically?

Reply:
you have to use `await` otherwise `DisposeAsync()` won't be awaited.

Reply:
Correction, if you don't use `await` it calls `Dispose()` instead of `DisposeAsync()`.